src/components/Checkbox/index.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useMemo, useCallback } from 'react';
+import {
+    MdCheckBox,
+    MdCheckBoxOutlineBlank,
+    MdIndeterminateCheckBox,
+} from 'react-icons/md';
+import {
+    randomString,
+    _cs,
+} from '@togglecorp/fujs';
+import { UiMode } from '../ThemeContext';
+import { useThemeClassName } from '../../hooks';
+
+import styles from './styles.css';
+
+export interface CheckboxProps {
+    className?: string;
+    labelClassName?: string;
+    checkIconClassName?: string;
+    label?: string | number;
+    disabled?: boolean;
+    readOnly?: boolean;
+    // NOTE: if value is false and indeterminate is true, show a filled checkbox
+    indeterminate?: boolean;
+    uiMode?: UiMode;
+    tooltip?: string;
+    value: boolean;
+    onChange: (value: boolean) => void;
+}
+
+const Checkbox = (props: CheckboxProps) => {
+    const {
+        label,
+        tooltip,
+        className: classNameFromProps,
+        value,
+        disabled,
+        readOnly,
+        onChange,
+        checkIconClassName,
+        labelClassName: labelClassNameFromProps,
+        indeterminate,
+        uiMode,
+        ...otherProps
+    } = props;
+
+    const inputId = useMemo(
+        () => randomString(16),
+        [],
+    );
+
+    const handleChange = useCallback(
+        (e) => {
```

Comment:
Don't we need to define the type of event?

Reply:
We should.